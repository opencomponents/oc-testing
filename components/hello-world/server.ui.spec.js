import 'babel-core/register';
import 'babel-polyfill';
import nightmare from 'nightmare';
import oc from 'oc';
import path from 'path';
import url from 'url';

// ui
const PORT = 3030;

const BASE_URL = url.format({
  protocol: process.env.PROTOCOL || 'http',
  hostname: process.env.HOST || 'localhost',
  port: process.env.PORT || PORT
});

const SUPERMAN = 'Superman';

test(`Hello ${SUPERMAN}`, async () => {
  const registry = new oc.Registry({
    baseUrl: `${BASE_URL}/`,
    local: true,
    port: PORT,
    hotReloading: false,
    path: path.resolve('./components'),
    env: { name: 'local' },
    verbosity: 0,
    discovery: true
  });

  registry.start();

  const page = nightmare({ show: true }).goto(`${BASE_URL}/hello-world/~preview?name=${SUPERMAN}`);
  const text = await page
    .wait(1000)
    .end()
    .evaluate(() => document.body.textContent);
  expect(text).toContain(`Hello ${SUPERMAN}`);
  expect(text).toMatchSnapshot();

  registry.close();
});
