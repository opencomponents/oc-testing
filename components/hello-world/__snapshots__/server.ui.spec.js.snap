// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Superman 1`] = `"Hello Superman"`;
